e2e/tests/negative/ha_volume_migration.robot

```diff
@@ -0,0 +1,397 @@
+    And Wait for volume 0 healthy
+    And Write data to volume 0
+
+    And Attach volume 0 to node 1
+    And Wait for volume 0 migration to be ready
+
+    # crash the engine on the migration node by killing its instance-manager pod
+    When Delete instance-manager on node 1
+    # longhorn retries migration and the volume is attached to both nodes eventually
+    Then Wait for volume 0 migration to be ready
+
+All Engines Crash During Migration
+    Given Create volume 0 with    migratable=True    accessMode=RWX    dataEngine=${DATA_ENGINE}
+    And Attach volume 0 to node 0
+    And Wait for volume 0 healthy
+    And Write data to volume 0
+
+    And Attach volume 0 to node 1
+    And Wait for volume 0 migration to be ready
+
+    # crash all engines by killing their instance-manager pods
+    When Delete instance-manager on node 0
+    And Delete instance-manager on node 1
+    And Delete instance-manager on node 2
+    # longhorn keeps the volume in detached state waiting for user to delete one of the 2 attachments
+    Then Wait for volume 0 detached
+
+    # detach the volume from one of nodes should make it fully attach to the other nod
+    When Detach volume 0 from node 1
+    Then Wait for volume 0 to stay on node 0
+    And Wait for volume 0 healthy
+    And Check volume 0 data is intact
+
+Volume Degraded Before Migration And Confirmation
+    Given Cordon node 2
+    And Delete instance-manager on node 2
+    And Create volume 0 with    migratable=True    accessMode=RWX    dataEngine=${DATA_ENGINE}
+    And Attach volume 0 to node 0
+    And Wait for volume 0 degraded
+    And Write data to volume 0
+
+    And Attach volume 0 to node 1
+    And Wait for volume 0 migration to be ready
+
+    # migration confirmation by detaching from the original node
+    When Detach volume 0 from node 0
+
+    Then Wait for volume 0 to migrate to node 1
+    And Wait for volume 0 degraded
+    And Check volume 0 data is intact
+
+Volume Degraded Before Migration And Rollback
+    Given Cordon node 2
+    And Delete instance-manager on node 2
+    And Create volume 0 with    migratable=True    accessMode=RWX    dataEngine=${DATA_ENGINE}
+    And Attach volume 0 to node 0
+    And Wait for volume 0 degraded
+    And Write data to volume 0
+
+    And Attach volume 0 to node 1
+    And Wait for volume 0 migration to be ready
+
+    # migration rollback by detaching from the migration node
+    When Detach volume 0 from node 1
+
+    Then Wait for volume 0 to stay on node 0
+    And Wait for volume 0 degraded
+    And Check volume 0 data is intact
+
+Volume Degraded Between Migration And Confirmation
+    Given Create volume 0 with    migratable=True    accessMode=RWX    dataEngine=${DATA_ENGINE}
+    And Attach volume 0 to node 0
+    And Wait for volume 0 healthy
+    And Write data to volume 0
+
+    And Attach volume 0 to node 1
+    And Wait for volume 0 migration to be ready
+
+    When Cordon node 2
+    And Delete instance-manager on node 2
```

Comment:
I have a question: Should we `wait for volume 0 to degrade` first before `detaching volume 0 from node 0`?

Reply:
I'm afraid that it could make test cases flaky because of a timing issue since it's not guaranteed the test code can catch the transient state. WDYT @roger-ryao?